elkodon_bb/posix/tests/signal_tests.rs

```diff
@@ -261,9 +262,11 @@ fn signal_termination_requested_with_terminate_works() {
 
     assert_that!(!SignalHandler::termination_requested(), eq true);
     assert_that!(Process::from_self().send_signal(Signal::Terminate), is_ok);
-    std::thread::sleep(std::time::Duration::from_millis(10));
 
-    assert_that!(SignalHandler::termination_requested(), eq true);
+    assert_that!(
+        || { SignalHandler::termination_requested() == true },
+        block_until_true
+    );
```

Comment:
Not sure if this is the right fix. A sleep of 1µs also fixes the problem ... at least on linux

Reply:
@elBoberido In my opinion it is the correct approach. If this fails and deadlocks the internal watchdog in assert_that will kill the process after 10 seconds